src/connectors/ConnectorWalletConnect.ts

```diff
@@ -7,32 +7,32 @@ export default async function init(): Promise<Connector> {
   )
   return {
     web3ReactConnector({
-      chainId,
-      rpcUrl,
+      rpc,
       bridge,
       pollingInterval,
     }: {
-      chainId: number
-      rpcUrl: string
+      rpc: RPC
```

Comment:
Maybe instead of creating a new type, pass ChainInformation here, and add an optional rpcUrl string to it?

Reply:
@WissenIstNacht opinion?

Reply:
I think It's not possible because type ChainInformation can't match with RPC input

Reply:
I think both approaches are feasible.

After some thought I think I'd stick to the the `RPC` type approach. The reason being that `chains.ts` contains static (hardcoded) information. Adding the `rpcUrl` to that data structure would mean that user's would have to modify the data structure and then use it for the connector.